main.tf

```
@@ -98,18 +112,43 @@ resource "aws_eks_cluster" "default" {
 
   dynamic "kubernetes_network_config" {
     for_each = local.use_ipv6 ? [] : compact([var.service_ipv4_cidr])
+
     content {
       service_ipv4_cidr = kubernetes_network_config.value
     }
   }
 
   dynamic "kubernetes_network_config" {
     for_each = local.use_ipv6 ? [true] : []
+
     content {
       ip_family = "ipv6"
     }
   }
 
+  dynamic "kubernetes_network_config" {
+    for_each = local.auto_mode_enabled ? [1] : []
+
+    content {
+      dynamic "elastic_load_balancing" {
+        for_each = local.auto_mode_enabled ? [1] : []
+        content {
+          enabled = local.auto_mode_enabled
+        }
+      }
+    }
+  }
+
+  dynamic "storage_config" {
+    for_each = local.auto_mode_enabled ? [1] : []
+
+    content {
+      block_storage {
+        enabled = local.auto_mode_enabled
```

Again, if required, then just use `true` to emphasize the fact. If not required, make it a separate input option.

```suggestion
        enabled = true
```